test/test_data/callback.rs

```diff
@@ -28,6 +28,23 @@ pub extern "C" fn run2(val: i32) -> i32 {
     }
 }
 
+#[no_mangle]
+pub extern "C" fn infinite_loop(val: i32) -> i32 {
+    let mut count = val;
+    loop {
+        count += 1;
+        if count >= 10 {
+            count -= 2;
+        }
+        if count >= 100 {
+            break;
+        }
+    }
+    unsafe {
+        callback2(val) + A
+    }
+}
```

Comment:
Again, there is no reason to make this complex. This simple infinite loop also works:
```suggestion
pub extern "C" fn infinite_loop() {
    let mut _count: u64 = 0;
    loop {
        _count += 1;
    }
}
```